Master.py

```diff
@@ -161,41 +186,152 @@ def generate_landsea(paths, param):
     n_high = param["n_high"]
     Crd_all = param["Crd_all"]
     res_desired = param["res_desired"]
+    res_weather = param["res_weather"]
     GeoRef = param["GeoRef"]
 
-    timecheck('Start_Land')
-    nRegions = param["nRegions_land"]
-    regions_shp = param["regions_land"]
-    Crd_regions = param["Crd_regions"][0:nRegions, :]
-    Ind = ind_merra(Crd_regions, Crd_all, res_desired)
+    timecheck('Start')
+    timecheck('Start Land')
+    # Read shapefile of scope
+    scope_shp = gpd.read_file(paths["spatial_scope"])
+    param["spatial_scope"] = define_spatial_scope(scope_shp)
+    ymax, xmax, ymin, xmin = Crd_all
+    bounds_box = Polygon([(xmin, ymin), (xmin, ymax), (xmax, ymax), (xmax, ymin)])
+
+    # Extract land areas
+    countries_shp = gpd.read_file(paths["Countries"], bbox=scope_shp)
+    countries_shp = countries_shp.to_crs({'init': 'epsg:4326'})
+    # Crop all polygons and take the part inside the bounding box
+    countries_shp['geometry'] = countries_shp['geometry'].intersection(bounds_box)
+    countries_shp = countries_shp[countries_shp.geometry.area > 0]
+    param["regions_land"] = countries_shp
+    param["nRegions_land"] = len(param["regions_land"])
+    Crd_regions_land = np.zeros((param["nRegions_land"], 4))
+    Ind = np.zeros((param["nRegions_land"], 4)).astype('int')
     A_land = np.zeros((m_high, n_high))
-
-    for reg in range(0, nRegions):
-        A_region = calc_region(regions_shp.iloc[reg], Crd_regions[reg, :], res_desired, GeoRef)
+    status = 0
+    for reg in range(0, param["nRegions_land"]):
+        # Show status bar
+        status = status + 1
+        sys.stdout.write('\r')
+        sys.stdout.write('Creating A_land ' + '[%-50s] %d%%' % (
+            '=' * ((status * 50) // param["nRegions_land"]), (status * 100) // param["nRegions_land"]))
+        sys.stdout.flush()
+        # Box coordinates for MERRA2 data
+        r = countries_shp.bounds.iloc[reg]
+        box = np.array([r["maxy"], r["maxx"], r["miny"], r["minx"]])[np.newaxis]
+        Crd_regions_land[reg, :] = crd_merra(box, res_weather)
+        Ind[reg, :] = ind_merra(Crd_regions_land[reg, :], Crd_all, res_desired)
```

Comment:
r, box, Crd_regions_land do not need to be calculated
Ind can be calculated in the top, out of the loop